File: sdks/csharp/rest/src/Brume.Rest/Types/PresenceResponse.cs
using Brume.Rest.Core;
using global::System.Text.Json;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

/// <summary>
/// `GET /v1/channels/:channel/presence` success response.
/// </summary>
[Serializable]
public record PresenceResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("channel")]
    public required string Channel { get; set; }

    [JsonPropertyName("connection_count")]
    public required int ConnectionCount { get; set; }

    /// <summary>
    /// Connection roster. Empty if presence tracking is disabled or the
    /// channel has no subscribers.
    /// </summary>
    [JsonPropertyName("presence")]
    public IEnumerable<ConnectionInfo> Presence { get; set; } = new List<ConnectionInfo>();

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/rest/src/Types/PresenceResponse.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;
use Brume\Rest\Core\Types\ArrayType;

/**
 * `GET /v1/channels/:channel/presence` success response.
 */
class PresenceResponse extends JsonSerializableType
{
    /**
     * @var string $channel
     */
    #[JsonProperty('channel')]
    public string $channel;

    /**
     * @var int $connectionCount
     */
    #[JsonProperty('connection_count')]
    public int $connectionCount;

    /**
     * Connection roster. Empty if presence tracking is disabled or the
     * channel has no subscribers.
     *
     * @var array<ConnectionInfo> $presence
     */
    #[JsonProperty('presence'), ArrayType([ConnectionInfo::class])]
    public array $presence;

    /**
     * @param array{
     *   channel: string,
     *   connectionCount: int,
     *   presence: array<ConnectionInfo>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->channel = $values['channel'];
        $this->connectionCount = $values['connectionCount'];
        $this->presence = $values['presence'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
